drivers/temperature/max31865pmb1/max31865.c

```diff
@@ -0,0 +1,361 @@
+ */
+void max31865_clear_fault(struct max31865_dev *device)
+{
+	uint8_t faultReg = 0x00;
+	max31865_read_raw(device, &faultReg);
+	faultReg &= ~0x2C;
+	faultReg |= 0x02;
+	uint8_t clrReg = 0x00;
+	max31865_write_raw(device, clrReg, faultReg);
+}
+
+/**
+ * @brief Enable the bias voltage on the RTD sensor
+ * @param device MAX31865 descriptor
+ * @param b If true bias is enabled, else disabled
+ */
+void max31865_enable_bias(struct max31865_dev *device, bool b)
+{
+	uint8_t confReg = 0x00;
+	uint8_t biasReg = 0x00;
+	max31865_read_raw(device, &confReg);
+
+	if (b)
+		confReg |= 0x80;
+	else
+		confReg &= ~0x80;
+
+	max31865_write_raw(device, biasReg, confReg);
+}
+
+/**
+ * @brief Option for continuous conversions between 50/60 Hz
+ * @param device MAX31865 descriptor
+ * @param b If true, auto conversion is enabled
+ */
+void max31865_auto_convert(struct max31865_dev *device, bool b)
+{
+	uint8_t confReg = 0x00;
+	uint8_t convReg = 0x00;
+	max31865_read_raw(device, &confReg);
+	if (b)
+		confReg |= 0x40;
+	else
+		confReg &= ~0x40;
+
+	max31865_write_raw(device, convReg, confReg);
+}
+
+/**
+ * @brief Option for 50Hz or 60Hz noise filters
+ * @param device MAX31865 descriptor
+ * @param b If true, 50Hz noise is filtered, else 60Hz(default)
+ */
+void max31865_enable_50Hz(struct max31865_dev *device, bool b)
+{
+	uint8_t confReg = 0x00;
+	uint8_t filtReg = 0x00;
+	max31865_read_raw(device, &confReg);
+
+	if (b)
+		confReg |= 0x01;
+	else
+		confReg &= ~0x01;
+
+	max31865_write_raw(device, filtReg, confReg);
+}
+
+/**
+ * @brief Write the lower and upper values into the threshold fault
+    register to values as returned by readRTD()
+ * @param device MAX31865 descriptor
+ * @param lower raw lower threshold
+ * @param upper raw upper threshold
+ */
+void max31865_set_threshold(struct max31865_dev *device, uint16_t lower,
+			    		uint16_t upper)
+{
+	uint16_t lsb;
+	uint16_t msb;
+
+	lsb = upper & 0xFF;
+	msb = upper >> 8;
+
+	max31865_write_raw(device, MAX31865_HFAULTMSB_REG, (uint8_t)lsb);
+	max31865_write_raw(device, MAX31865_HFAULTLSB_REG, (uint8_t)msb);
+
+	lsb = lower & 0xFF;
+	msb = lower  >> 8;
+
+	max31865_write_raw(device, MAX31865_LFAULTMSB_REG, (uint8_t)lsb);
+	max31865_write_raw(device, MAX31865_LFAULTLSB_REG, (uint8_t)msb);
+}
+
+/**
+ * @brief Read the raw 16-bit lower threshold value
+ * @param device MAX31865 descriptor
+ * @return The raw unsigned 16-bit value, NOT temperature value
+ */
+uint16_t max31865_get_lower_threshold(struct max31865_dev *device)
+{
+	uint8_t lowmsb = 0x05;
+	uint8_t lowlsb = 0x06;
+	max31865_read_raw(device, &lowmsb);
+	max31865_read_raw(device, &lowlsb);
+
+	uint16_t lowthresh = 0x00;
+
+	lowthresh = (((uint16_t)lowmsb << 8) | (uint16_t)lowlsb);
+
+	return lowthresh;
+}
+
+/**
+ * @brief Read the raw 16-bit upper threshold value
+ * @param device MAX31865 descriptor
+ * @return The raw unsigned 16-bit value, NOT temperature value
+ */
+uint16_t max31865_get_upper_threshold(struct max31865_dev *device)
```

Comment:
ditto: use a pointer variable in the function prototype to store the threshold. return afferent error code from the read/write functions.